src/ring_theory/algebraic_independent.lean

```diff
@@ -0,0 +1,434 @@
+/-
+Copyright (c) 2021 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import ring_theory.adjoin.basic
+import linear_algebra.linear_independent
+import ring_theory.mv_polynomial.basic
+import data.mv_polynomial.supported
+import ring_theory.algebraic
+import data.mv_polynomial.equiv
+/-!
+# Algebraic Independence
+
+This file defines algebraic independence of a family of element of an `R` algebra
+
+## Main definitions
+
+* `algebraic_independent` - `algebraic_independent R v` states the family of vectors `v`
+  is algebraically independent over `R`, meaning that the canonical map out of the multivariable
+  polynomial ring is injective.
+
+* `algebraic_independent.repr` - The canonical map from the subalgebra generated by an
+  algebraic independent family into the polynomial ring.
+
+## References
+
+* [Stacks: Transcendence](https://stacks.math.columbia.edu/tag/030D)
```

Comment:
I don't fully get the notation at this link; if `k[Xi;i∈I]` is the field `k` extended with the elements `Xi`, does it indicate that the `Xi` must commute?

Reply:
Yes